From 8255cdb251e670fef23fa47dfd753e096457aa02 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Thu, 14 Mar 2019 15:39:03 -0700
Subject: [PATCH 18/31] UBUNTU: SAUCE: Revert "apparmor: modify audit rule
 support to support profile stacks"

Revert currently unused by Ubuntu apparmor features that require secid
support, to enable apparmor LSM stacking with the 5.1 LSM stacking
patchset.

This reverts commit 2ab47dae54d567bbb1ad3e96e5b2601cc13f4d2b.

Signed-off-by: John Johansen <john.johansen@canonical.com>
Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
---
 security/apparmor/audit.c | 27 +++++++++++++++++----------
 1 file changed, 17 insertions(+), 10 deletions(-)

diff --git a/security/apparmor/audit.c b/security/apparmor/audit.c
index 966e796459fe..060fe1691fea 100644
--- a/security/apparmor/audit.c
+++ b/security/apparmor/audit.c
@@ -161,7 +161,7 @@ int aa_audit(int type, struct aa_profile *profile, struct common_audit_data *sa,
 }
 
 struct aa_audit_rule {
-	struct aa_label *label;
+	char *profile;
 };
 
 void aa_audit_rule_free(void *vrule)
@@ -169,8 +169,7 @@ void aa_audit_rule_free(void *vrule)
 	struct aa_audit_rule *rule = vrule;
 
 	if (rule) {
-		if (!IS_ERR(rule->label))
-			aa_put_label(rule->label);
+		kfree(rule->profile);
 		kfree(rule);
 	}
 }
@@ -193,11 +192,13 @@ int aa_audit_rule_init(u32 field, u32 op, char *rulestr, void **vrule)
 	if (!rule)
 		return -ENOMEM;
 
-	/* Currently rules are treated as coming from the root ns */
-	rule->label = aa_label_parse(&root_ns->unconfined->label, rulestr,
-				     GFP_KERNEL, true, false);
-	if (IS_ERR(rule->label))
-		return PTR_ERR(rule->label);
+	rule->profile = kstrdup(rulestr, GFP_KERNEL);
+
+	if (!rule->profile) {
+		kfree(rule);
+		return -ENOMEM;
+	}
+
 	*vrule = rule;
 
 	return 0;
@@ -223,6 +224,8 @@ int aa_audit_rule_match(u32 sid, u32 field, u32 op, void *vrule)
 {
 	struct aa_audit_rule *rule = vrule;
 	struct aa_label *label;
+	struct label_it i;
+	struct aa_profile *profile;
 	int found = 0;
 
 	label = aa_secid_to_label(sid);
@@ -230,8 +233,12 @@ int aa_audit_rule_match(u32 sid, u32 field, u32 op, void *vrule)
 	if (!label)
 		return -ENOENT;
 
-	if (aa_label_is_subset(label, rule->label))
-		found = 1;
+	label_for_each(i, label, profile) {
+		if (strcmp(rule->profile, profile->base.hname) == 0) {
+			found = 1;
+			break;
+		}
+	}
 
 	switch (field) {
 	case AUDIT_SUBJ_ROLE:
-- 
2.25.1

